src/ElementalChat.vue

```diff
@@ -215,30 +215,40 @@ export default {
     handleToDisplay () {
       return this.isHoloAnonymous ? 'anonymous' : this.agentHandle
     },
-    shouldAutoSignIn () {
-      // Note: isAnonymousEnabled refers to whether the feature is enabled when the app is built. isHoloAnonymous refers to the current status of our hosted agent.
-      return isHoloHosted() && !isAnonymousEnabled() && this.holoStatus === 'ready' && this.isHoloAnonymous
+    shouldHandleLogin() {
+      return isHoloHosted() && this.holoStatus === 'ready' && this.isHoloAnonymous === true
     },
     canMakeZomeCalls() {
+      // Note: isAnonymousEnabled refers to whether the feature is enabled when the app is built. isHoloAnonymous refers to the current status of our hosted agent.
       return isHoloHosted() ? this.holoStatus === 'ready' && (isAnonymousEnabled() || this.isHoloAnonymous === false) : !this.conductorDisconnected
     }
   },
   created () {
     console.log('agentKey started as', this.agentKey)
   },
   watch: {
-    async shouldAutoSignIn (should) {
+    async shouldHandleLogin (should) {
+      console.log(`watcher activated: shouldHandleLogin=${should}`)
       if (should) {
-        await this.holoLogin()
+        const urlParams = new URLSearchParams(window.location.search)
+        if (urlParams.has('signin')) {
+          this.$store.dispatch('holochain/holoSignin')
+        } else if (urlParams.has('signup') || !isAnonymousEnabled()) { 
+          this.$store.dispatch('holochain/holoSignup')
+        }
       }
```

Comment:
I'm wondering what happens when I hit the back arrow on the login screen now. It will take me back to EC UI, I won't have logged in, and I'll be anonymous (even if `isAnonymousEnabled() === false`). How can we prevent that? I'll test locally what happens if I introduce a `while (should) ` into the watcher

Reply:
Update: I added `cancellable: false` so the back arrow won't be a problem